src/components/popupWithForm.js

```diff
@@ -20,7 +21,9 @@ export default class PopupWithForm extends Popup {
   }
 
   setEventListeners() {
+    //modify to work with modals without inputs
     super.setEventListeners();
+    //
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault(); //prevent page refresh
       this._handleFormSubmit(this._getInputValues()); //call handleFormSubmit
```

Comment:
Please, delete `this._popupForm.reset();` from line 30

You need to **reset** the form only  in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
You need to **reset** the form only  in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.